aictl/cli/main.py

```diff
@@ -112,6 +112,36 @@ def ip2p(args):
     print("### saving image files")
     output.images[0].save(args.output_path)
 
+def t2v(args):
+    import torch
+    from diffusers import DiffusionPipeline
+    from diffusers.utils import export_to_video
+
+     # check if on mac and mps is available, fallback to cuda then cpu
+    is_mac = False
+    device = "cuda:0" if torch.cuda.is_available() else "cpu"
+    if torch.backends.mps.is_available():
+        is_mac = True
+        device = torch.device("mps")
+        print("MPS device detected. Using MPS.")
+
+    pipe = DiffusionPipeline.from_pretrained("damo-vilab/text-to-video-ms-1.7b", torch_dtype=torch.float16, variant="fp16")
```

Comment:
float16 will not work on Mac so what I did for the other commands was to add this line to switch to float32 when a Mac is detected:
```python
model_type = torch.float32 if is_mac else torch.float16
```
Then you would change this line to:
```python
pipe = DiffusionPipeline.from_pretrained("damo-vilab/text-to-video-ms-1.7b", torch_dtype=model_type)
```